analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.ExceptionCandidate.cs

```diff
@@ -0,0 +1,516 @@
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_DynamicInvocation()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+dynamic x = new Object();
+try
+{
+    tag = ""InTry"";
+    int y = x.Invocation();
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_DynamicMemberReference()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+dynamic x = new Object();
+try
+{
+    tag = ""InTry"";
+    int y = x.Property;
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_DynamicObjectCreation()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    var x = new Sample(d);
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code, ", dynamic d").Validator;
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_EventReference_Instance()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    p.Event += (sender, args) => { };
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+            var validator = SETestContext.CreateCS(code, ", Person p").Validator;
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasUnknownException(x));
+            validator.ExitStates.Should().ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_EventReference_Static()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    StaticEvent += (sender, args) => { };
+}
+catch
+{
+    tag = ""InCatch"";
```

Comment:
To prevent us from updating the UT in case something goes wrong.
```suggestion
    tag = ""UnreachableInCatch"";
```

Reply:
This can be done on few other places